### ISSUE TITLE
Reuse existing boxes when editing

### ISSUE THREAD

**Issue by MikkelPaulson** [initiative-sh/initiative.sh#190]
* When editing the Thing that was most recently displayed, update the existing block rather than creating a new one.
* Highlight the changed line(s).

**Comment by ChrisRenfrow** [comment]
@MikkelPaulson 

I am interested in implementing this, and after taking a look around I came up with a couple approaches.

Approach no.1 could involve modifying `/core/src/world/*/view.rs` to populate the `id` attribute of each `.thing-box` with a unique value derived from the thing's UUID. Then, in the front-end we can try and select the matching element in the document and, if it exists and is "in view", replace the existing element.

A less invasive option might be to assign a unique id derived from the name of the thing (`h1` contents) in the front-end, amending the output (if it's of class `.thing-box`) with the derived id before inserting into the document and performing the final step from the other approach.

Which of these sound more appealing to you? Is there another path to consider?

**Comment by MikkelPaulson** [comment]
@ChrisRenfrow Sounds good, go for it!

I had something like the first approach in mind. There's no need to hack it on as a frontend-only change. However, be mindful that the `id` field is supposed to be unique to a DOM; using `uuid` (or something else) instead would be better.

I would only update the *most recent* box, rather than anything still visible on the screen. And I suggest a CSS animation to show that a field has been changed, maybe highlighting it for a second or something. *That* diff can be done client (JS)-side; trying to get that information across the WASM FFI would be a pain.

**Comment by ChrisRenfrow** [comment]
Oh, I think I understand, you're asking that I use some arbitrary property like `data-uuid` instead of `id`? That works for me. And you answered my next line of questioning, so I think I'm set to begin. :)

**Comment by MikkelPaulson** [comment]
Exactly. Because the same entity can appear multiple times on a page, we just don't want it to appear consecutively anymore.
